.github/workflows/docs.yml

```diff
@@ -81,20 +81,14 @@ jobs:
           cd build/html
           touch .nojekyll
           git add -A --force .
-          git commit -m "Automated documentation rebuild"
+          git diff --exit-code --quiet || git commit -m "Automated documentation rebuild"  # only commit if changes were made
```

Comment:
It seems like if a new file is **added** rather than modified, we need to add the `--staged` flag to `git diff`